bcipy/tasks/query_selection.py

```diff
@@ -0,0 +1,190 @@
+            reset():
+                this selection method is memoryless, therefore this is void\
+            update_query():
+                updates the querying method and picks the particular query
+        """
+
+    def __init__(self, len_alp, len_query=4):
+        self.alp = list(range(len_alp))
+        self.len_query = len_query
+
+    def reset(self):
+        """ This object does not require history. Reset is a void function. """
+        tmp = None
+
+    def update_query(self, p):
+        """ with the final belief over the system, updates the querying method and
+            generates len_query most likely queries.
+            Args:
+                p(list[float]): list of probability distribution over state
+                    estimates
+            Return:
+                pos_query(list[int]): positions of queries in alphabet """
+        tmp = [i for i in self.alp]
+        pos_query = best_selection(tmp, p, self.len_query)
+
+        return pos_query
+
+
+class MomentumQuerying(object):
+    """ picks arms in multi armed bandit(MAB) with the momentum reward.
+        reward is convex combination of the momentum of a particular query and
+        the entropy. given the rewards makes greedy arm selection.
+        Attr:
+            len_alp(int): Length of the alphabet. Query selection returns
+                integers as queries, each corresponding to the location of the
+                query to be selected
+            lam(float): in [0,1] convex combination parameter
+            gam(float): in [0,1] history decay rate for momentum
+            momentum(ndarray[float]): momentum values for each element
+            prob_history(list[list[float]]): probability values for each step
+                in the history for each element in the alphabet
+            last_query(list[str]): final query generated by the system
+        Functions:
+            reset():
+                resets the system memory
+                """
+
+    def __init__(self, len_alp, len_query=4, gam=.9, lam=.9):
+        self.alp = list(range(len_alp))
+        self.lam = lam
+        self.gam = gam
+        self.len_query = len_query
+        self.momentum = np.zeros(len(self.alp))
+        self.prob_history = []
+        self.last_query = []
+
+    def reset(self, lam=.9):
+        """ resets the history related items in the query method
+            Args:
+                lam(float): in [0,1] convex combination parameter """
+        self.lam = lam
+        self.momentum = np.zeros(len(self.alp))
+        self.prob_history = []
+
+    def update_query(self, p):
+        """ with the final belief over the system, updates the querying method
+            and generates len_query most likely queries.
+            Args:
+                p(list[float]): list of probability distribution over the state
+                    estimates
+            Return:
+                pos_query(list[int]): positions of queries in alphabet """
+        tmp = p[:]
```

Comment:
Rather than `tmp`, maybe call this something like `probabilities`?

Reply:
`tmp` appears due to assignment problem in python. This is literally a temporary element in order to avoid overwriting of the original element. Therefore, I prefer to keep it as is.

Reply:
He's saying to give it a name to help readers understand what you are assigning.